specification/contosowidgetmanager/Contoso.Management/employee.tsp

```diff
@@ -0,0 +1,63 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using Azure.Core;
+using Azure.ResourceManager;
+
+namespace Microsoft.Contoso;
+
+/** Employee resource */
+model Employee is TrackedResource<EmployeeProperties> {
+  ...ResourceNameParameter<Employee>;
+}
+
+/** Employee properties */
+model EmployeeProperties {
+  /** Age of employee */
+  age?: int32;
+
+  /** City of employee */
+  city?: string;
+
+  /** Profile of employee */
+  @encode("base64url")
+  profile?: bytes;
+
+  /** The status of the last operation. */
+  @visibility("read")
+  provisioningState?: ProvisioningState;
+}
+
+/** The resource provisioning state. */
+@lroStatus
+union ProvisioningState {
+  ResourceProvisioningState,
+
+  /** The resource is being provisioned */
+  Provisioning: "Provisioning",
+
+  /** The resource is updating */
+  Updating: "Updating",
+
+  /** The resource is being deleted */
+  Deleting: "Deleting",
+
+  /** The resource create request has been accepted */
+  Accepted: "Accepted",
+
+  string,
+}
+
+@armResourceOperations
+interface Employees {
+  get is ArmResourceRead<Employee>;
+  createOrUpdate is ArmResourceCreateOrReplaceAsync<Employee>;
+  update is ArmResourcePatchSync<Employee, EmployeeProperties>;
```

Comment:
we should not use this template for now.

Reply:
Good catch.  could you replace it with ArmCustomPatchSync?

Reply:
@allenjzhang , should we update this in both of this PR and `tsp init` template project?

Reply:
Just this  `update is ArmCustomPatchSync<Employee>;` or do we want more custom `Employee` patch model?

Reply:
Similar like this change  https://github.com/Azure/azure-rest-api-specs/pull/29834/files
```
  update is ArmCustomPatchSync<
    Employee,
    Azure.ResourceManager.Foundations.ResourceUpdateModel<
      Employee,
      EmployeeProperties
    >
  >;
```
Maybe we should also add emit-common-types-schema: "never" in typespec-autorest.

Reply:
submited here https://github.com/Azure/azure-rest-api-specs/pull/30040/commits/91815e97dd1937918c525b31397174c9002ca0ac I think it should be good to go now.